api/providers/SupplierProvider.cs

```diff
@@ -0,0 +1,50 @@
+using DTO.Supplier;
+using FluentValidation;
+using Microsoft.EntityFrameworkCore;
+
+public class SupplierProvider : BaseProvider<Supplier>
+{
+    private ContactProvider _contactProvider;
+    private AddressProvider _addressProvider;
+    private IValidator<Supplier> _supplierValidator;
+
+    public SupplierProvider(AppDbContext db, IValidator<Supplier> validator, ContactProvider contactProvider, AddressProvider addressProvider) : base(db)
+    {
+        _supplierValidator = validator;
+        _addressProvider = addressProvider;
+        _contactProvider = contactProvider;
+    }
+
+    public override Supplier? Create(BaseDTO createValues)
+    {
+        SupplierRequest? req = createValues as SupplierRequest;
+        var allSuppliers = _db.Suppliers.Include(c => c.Contact).Include(a => a.Address).ToList();
+
+        if(allSuppliers.Any(s => s.Reference == req.Reference)) throw new ApiFlowException("Could not process create supplier request. Supplier with this reference code already exists.");
```

Comment:
Is hiervoor een requirement wat de PO aangaf? Waar hij wilde dat er een check is dat alle supplier codes uniek zijn?